parts/linux/cloud-init/artifacts/validate-kubelet-credentials.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -euo pipefail
+
+# this gives us logs_to_events and retry wrappers
+source /opt/azure/containers/provision_source.sh 
+
+KUBECONFIG_PATH="${KUBECONFIG_PATH:-/var/lib/kubelet/kubeconfig}"
+BOOTSTRAP_KUBECONFIG_PATH="${BOOTSTRAP_KUBECONFIG_PATH:-/var/lib/kubelet/bootstrap-kubeconfig}"
+
+VALIDATE_KUBELET_CREDENTIALS_MAX_RETRIES=${VALIDATE_KUBELET_CREDENTIALS_MAX_RETRIES:-10}
+VALIDATE_KUBELET_CREDENTIALS_RETRY_DELAY_SECONDS=${VALIDATE_KUBELET_CREDENTIALS_RETRY_DELAY_SECONDS:-3}
+VALIDATE_KUBELET_CREDENTIALS_RETRY_TIMEOUT_SECONDS=${VALIDATE_KUBELET_CREDENTIALS_RETRY_TIMEOUT_SECONDS:-5}
+
+function validateKubeconfig {
+    local kubeconfig_path=$1
+
+    if ! retrycmd_if_failure $VALIDATE_KUBELET_CREDENTIALS_MAX_RETRIES \
+        $VALIDATE_KUBELET_CREDENTIALS_RETRY_DELAY_SECONDS \
+        $VALIDATE_KUBELET_CREDENTIALS_RETRY_TIMEOUT_SECONDS \
+        kubectl auth whoami --kubeconfig "$kubeconfig_path"; then
+
+        echo "kubelet credential validation failed"
+        exit 1
+    fi
+}
+
+function validateKubeletCredentials {
+    if [ ! -f "$KUBECONFIG_PATH" ] && [ ! -f "$BOOTSTRAP_KUBECONFIG_PATH" ]; then
+        echo "both kubeconfig: $KUBECONFIG_PATH and bootstrap-kubeconfig: $BOOTSTRAP_KUBECONFIG_PATH do not exist, unable to start kubelet"
+        exit 1
+    fi
+
+    if ! which kubectl; then
+        echo "kubectl not found, will skip kubelet credential validation"
+        exit 0
+    fi
+
+    if [ -f "$KUBECONFIG_PATH" ]; then
```

Comment:
do we need this condition since we are checking above that both Kubeconfig file exists

Reply:
yes, the first condition above checks that _at least_ one of kubeconfig or bootstra-kubeconfig exists, not that both exist